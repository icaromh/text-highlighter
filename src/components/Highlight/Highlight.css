.Hightlight {
  padding: 5px;
  font-size: 1rem;
  line-height: 1.5;
}
