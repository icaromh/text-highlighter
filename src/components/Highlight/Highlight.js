import React from 'react'
import PropTypes from 'prop-types'

const Highlight = ({ data }) => (
  <mark className="Hightlight" style={{ backgroundColor: data.color }}>
    {data.sentence}
  </mark>
)

Highlight.propTypes = {
  data: PropTypes.shape({
    sentence: PropTypes.string.isRequired,
    color: PropTypes.string.isRequired,
    head: PropTypes.number.isRequired,
    tail: PropTypes.number.isRequired,
  }).isRequired,
}

export default Highlight
