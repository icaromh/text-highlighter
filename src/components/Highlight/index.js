import Highlight from './Highlight'
import './Highlight.css'


export default Highlight
