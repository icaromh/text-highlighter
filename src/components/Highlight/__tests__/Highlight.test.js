import React from 'react'
import { shallow } from 'enzyme'

import Highlight from '../index'

describe('#Highlight', () => {
  test('expect to render', () => {
    const dummyData = {
      sentence: 'I will travel across the land',
      color: 'red',
    }
    const wrapper = shallow(<Highlight data={dummyData} />)
    expect(wrapper).toMatchSnapshot()
  })
})
