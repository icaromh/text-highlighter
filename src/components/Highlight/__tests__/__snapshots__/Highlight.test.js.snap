// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#Highlight expect to render 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Highlight
    data={
      Object {
        "color": "red",
        "sentence": "I will travel across the land",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "I will travel across the land",
      "className": "Hightlight",
      "style": Object {
        "backgroundColor": "red",
      },
    },
    "ref": null,
    "rendered": "I will travel across the land",
    "type": "mark",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I will travel across the land",
        "className": "Hightlight",
        "style": Object {
          "backgroundColor": "red",
        },
      },
      "ref": null,
      "rendered": "I will travel across the land",
      "type": "mark",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
